dwm1001/examples/lis2dh12.rs

```diff
@@ -1,43 +1,77 @@
 //! Accesses the LIS2DH12 3-axis accelerometer
+//!
+//! To build:
+//! cargo build --example lis2dh12 --features "dev rt semihosting" --release
+//!
+//! To debug:
+//! openocd -f openocd_jlink.cfg
+//! arm-none-eabi-gdb target/thumbv7em-none-eabihf/release/examples/lis2dh12 -x debug.gdb
 
 #![no_main]
 #![no_std]
 
-
 extern crate panic_semihosting;
 
-
 use cortex_m_rt::entry;
 
-use dwm1001::{
-    debug,
-    DWM1001,
-    print,
-};
+use dwm1001::{debug, print, DWM1001};
 
+use lis2dh12::{self, Accelerometer};
 
 #[entry]
 fn main() -> ! {
     debug::init();
 
-    let mut dwm1001 = DWM1001::take().unwrap();
+    print!("start\n");
+
+    let dwm1001 = DWM1001::take().unwrap();
 
     // SDO/SA0 is connected to the supply voltage, so the least significant bit
     // is 1. See datasheet, section 6.1.1.
-    let address = 0b0011001;
 
-    let mut write_buf = [0u8; 1];
-    let mut read_buf  = [0u8; 1];
+    let address = lis2dh12::SlaveAddr::Alternative(true);
+
+    let mut lis2dh12 =
+        lis2dh12::Lis2dh12::new(dwm1001.LIS2DH12, address).expect("lis2dh12 new failed");
```

Comment:
This could be done in `lib.rs`, then the user wouldn't have to do it themselves. But it looks like the constructor is also doing initialization (since it returns an error), so that would add a runtime cost for everyone. We could avoid that by adding the driver as an `Option<...>` and initializing it on-demand, or by patching the driver to have a separate `init` method.

Reply:
I prefer keeping it as is for the moment.